pretext/server.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from http.server import SimpleHTTPRequestHandler
+import logging
+import os
+from pathlib import Path
+import socketserver
+import typing as t
+import psutil
+
+from pretext.utils import hash_path, home_path
+
+# Get access to logger
+log = logging.getLogger("ptxlogger")
+
+
+@dataclass
+class RunningServerInfo:
+    """A simple dataclass to hold the information in the running servers file."""
+
+    pathHash: str
+    pid: int
+    port: int
+    binding: str
+
+    @staticmethod
+    def fromFileLine(line: str) -> RunningServerInfo:
```

Comment:
Note that methods in Python should be `snake_case`: https://peps.python.org/pep-0008/#method-names-and-instance-variables

Reply:
It's funny that the logging module doesn't follow that convention.  I've definitely been bad at following the convention, as my mind really doesn't like it at all.
Would it perhaps be possible to add this as something reported by the styling tools?   https://github.com/PyCQA/pep8-naming

Reply:
I think Java mixes `snake_case` and `camelCase`, if I recall? I know it's a convention elsewhere, but I try to stick with pep8 for Python, even if Python itself violates it in some core libraries...

And yeah, really this should be caught by our linter.